src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleKey.java

```diff
@@ -79,35 +81,44 @@ public final String toString() {
    *
    * <p>This method must not be called if the module has a {@link NonRegistryOverride}.
    */
-  public RepositoryName getCanonicalRepoNameWithVersion() {
-    return getCanonicalRepoName(/* includeVersion= */ true);
+  public RepositoryName getCanonicalRepoNameWithVersion(StarlarkSemantics semantics) {
+    return getCanonicalRepoName(/* includeVersion= */ true, semantics);
+  }
+
+  public RepositoryName getCanonicalRepoNameWithVersionForTesting() {
+    return getCanonicalRepoNameWithVersion(StarlarkSemantics.DEFAULT);
   }
 
   /**
    * Returns the canonical name of the repo backing this module, excluding its version. This name is
    * only guaranteed to be unique when there is a single version of the module in the entire dep
    * graph.
    */
-  public RepositoryName getCanonicalRepoNameWithoutVersion() {
-    return getCanonicalRepoName(/* includeVersion= */ false);
+  public RepositoryName getCanonicalRepoNameWithoutVersion(StarlarkSemantics semantics) {
+    return getCanonicalRepoName(/* includeVersion= */ false, semantics);
+  }
+
+  public RepositoryName getCanonicalRepoNameWithoutVersionForTesting() {
```

Comment:
Could this use `@VisibleForTesting` for enforcement?

Reply:
Done